File: client/src/types/book.ts
export type BookStatus = "Reading" | "Completed" | "Wishlist";

export interface Book {
  id: string;
  title: string;
  author: string;
  genre: string;
  status: BookStatus;
  rating?: number;
  review?: string;
  dateCompleted?: string;
  coverImageUrl?: string;
}

export interface BookDetailsDTO {
  id: string;
  title: string;
  author: string;
  genre: string;
  status: BookStatus;
  rating?: number;
  review?: string;
  dateAdded: string;
  dateCompleted?: string;
  isFavourite: boolean;
  coverImageUrl?: string;
}

export interface PaginatedResponse<T> {
  items: T[];
  totalItems: number;
  currentPage: number;
  totalPages: number;
}

File: server/BookTracker.Api/Models/Book.cs
using System.ComponentModel.DataAnnotations;

public enum BookStatus { Reading, Completed, Wishlist }

public class Book {
    public Guid Id { get; set; }

    [Required]
    public string Title { get; set; } = null!;

    [Required]
    public string Author { get; set; } = null!;

    public string Genre { get; set; } = null!;

    public BookStatus Status { get; set; }

    [Range(1, 5)]
    public int? Rating { get; set; }

    public string? Review { get; set; }

    public string? CoverImageUrl { get; set; }

    public DateTime DateAdded { get; set; } = DateTime.UtcNow;

    public DateTime? DateCompleted { get; set; }

    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
}
